### DISCUSSION TITLE
Is it possible to build Cardinal in a virtual machine in Windows?

### DISCUSSION THREAD

**Discussion by benaldhouse** [DISTRHO/Cardinal#542] [Q&A]
This is a very long shot, I'm guessing, since the wording in BUILDING.md looks fairly robust but would this be possible? If so, which VM software and VM might be worth trying?
Thank you for your patience reading this...

**Comment by falkTX** [comment]
sure yes, that is basically what the CI workers via github actions do.

See https://github.com/DISTRHO/Cardinal/blob/main/.github/workflows/build.yml#L401 for how the CI builds handle cross-compilation for windows, just replace `${{ matrix.target }}` with `win64` and ignore the steps with `${{ matrix.target == 'win32' }}` condition

**Reply by benaldhouse** [reply]
Thanks!

**Reply by falkTX** [reply]
note that if you dont care about having matching builds to the releases, you can skip the "pawpaw" stuff that takes a while to build, it is used for extra audio file formats and carla (plugin host), but not for any of the common modules.

**Reply by benaldhouse** [reply]
Thanks again, Felipe. I've installed an Ubuntu virtual machine on my laptap. It will take me some time setting up a programming environment on it and learning how to use it. I have programmed in C++ in Visual Studio on Windows before but not on Linux. I'm looking forward to broadening my horizons...

**Reply by benaldhouse** [reply]
Hi Filipe. I am very happy that I have now compiled Cardinal in Ubuntu. I thought I'd stay in Ubuntu for the time being since it has taken me a while to get this far and cross compiling for Windows seems like another fairly big step for me.  I am however having trouble starting the compiled program. It looks like something to do with communicating with JACK. Is there something I can do in JACK to fix this? I'm on Ubuntu 22.04 (Virtual Machine) and this is the error message I had...
Thanks,
Ben
`libjack.so.0 loaded successfully!
Cannot read socket fd = 9 err = Success
CheckRes error
JackSocketClientChannel read fail
Cannot open Cardinal client
JackShmReadWritePtr1::~JackShmReadWritePtr1 - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
Failed to create the JACK client, reason was:
Communication error with the JACK server.
`

**Reply by falkTX** [reply]
the JACK/main binary needs jackaudio.org stuff to be installed and running, you likely want to use the CardinalNative standalone instead

**Reply by benaldhouse** [reply]
I think there is something up with the audio in my virtual machine. I have managed get Cardinal running now but there's no sound but there isn't any sound from anywhere else in the virtual machine,  either.  I'm sorry I have bothered you with this, I clearly have problems outside of Cardinal.
